spec/controllers/challenge_signup_controller_spec.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+require 'spec_helper'
+
+RSpec.describe ChallengeSignupsController, type: :controller do
+  include LoginMacros
+  include RedirectExpectationHelper
+  let(:user) { create(:user) }
+  let(:signup) do
+    challenge = create(:gift_exchange, :closed)
+    collection = create(:collection, challenge: challenge)
+    signup = build(:challenge_signup, collection_id: collection)
+    signup.collection = collection
+    signup.save
+    signup
+  end
+  let(:collection) { signup.collection }
+  let(:challenge) { signup.collection.challenge }
+  let(:collection_owner) { User.find(collection.all_owners.first.user_id) }
+  let(:signup_owner) { Pseud.find(signup.pseud_id).user }
+
+  let(:open_signup) do
+    challenge = create(:gift_exchange, :open)
+    collection = create(:collection, challenge: challenge)
+    signup = build(:challenge_signup, collection_id: collection)
+    signup.collection = collection
+    signup.save
+    signup
+  end
+  let(:open_collection) { open_signup.collection }
+  let(:open_challenge) { open_signup.collection.challenge }
+  let(:open_signup_owner) { Pseud.find(open_signup.pseud_id).user }
+
+  describe "new" do
+    it "directs and errors if sign-up is not open" do
```

Comment:
Nothing that needs taking care of until this gets proper review, but typo: should be "redirects"